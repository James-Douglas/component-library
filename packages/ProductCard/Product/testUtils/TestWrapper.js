import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
