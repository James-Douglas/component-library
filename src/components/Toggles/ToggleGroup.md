# ToggleGroup component
The CTM ToggleGroup component, wraps a given set of toggles.

## Usage
~~~js
<ToggleGroup 
  label={label}
  id={id}
  name={name}
  tooltip={tooltip}
  handleChange={handleChange}
  selectedValue={selectedValue}
>
  <Toggle />
  <Toggle />
  <Toggle />
</ToggleGroup>
~~~

*Priority: picture > icon

## Props
The ToggleGroup component accepts the following props:

1. `name` *(required) string*
    * Name property to be passed to the toggles - required for radio buttons

7. `onToggle` *(required) function*
    * onChange handler function, called on select of a toggle with { id: <selected toggle id>, value: <selected toggle value> }

2. `label` *(optional) string*
    * Label for the ToggleGroup. Defaults to an empty string.

3. `tooltip` *(optional) object*
   * Tooltip object (see Tooltip documentation)
       
4. `forceFullWidth` *(optional) boolean* 
   * Forces the ToggleGroup to expand to 12 columns (default true for ToggleGroup)

5. `validationMessage` *(optional string)*
    * Displays given validation message and invalid styles on the component when provided.  
    
6. `id` *(optional) string*
    * Unique identifier for the toggle group
    
8. `rectOptions` *(optional) object*
    * Options object for rectangular toggles (see Toggle documentation)
    
9. `selectedValue` *(optional) string/number*
    * Value of the currently selected toggle (use to pre-select)
    
