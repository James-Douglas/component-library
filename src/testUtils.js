// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
