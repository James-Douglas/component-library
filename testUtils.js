import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
