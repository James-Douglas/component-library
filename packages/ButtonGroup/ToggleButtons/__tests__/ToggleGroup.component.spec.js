import React from 'react';
import { render, fireEvent } from '../../../../testUtils';
import ToggleGroup, { getChildren } from '../ToggleGroup.component';
import TextToggle from '../Text/TextToggle.component';
import 'jest-styled-components';

describe('getChildren()', () => {
  const tempID = 'temp-id';
  const getTestChildren = (data) => data.map((child) => React.cloneElement(
    <TextToggle id={tempID} title="test-label" value="test-value" />,
    { ...child },
  ));
  const didToggleCb = jest.fn();
  const clickCb = jest.fn();
  it('returns cloned children with base props added', () => {
    const testChildren = getTestChildren([
      {
        label: 'test toggle a',
        id: 'a',
        value: 'val-a',
      },
    ]);
    const result = getChildren(
      'group-id',
      testChildren,
      'test-group',
      null,
      didToggleCb,
      clickCb,
      null,
      '10rem',
      '10rem',
    );
    const {
      name,
      id,
      selectedValue,
      handleToggle,
      handleClick,
      value,
      contentWidth,
      contentHeight,
    } = result[0].props;

    expect(name).toEqual('test-group');
    expect(value).toEqual('val-a');
    expect(id).toEqual('a');
    expect(selectedValue).toEqual(null);
    expect(handleToggle).toEqual(didToggleCb);
    expect(handleClick).toEqual(clickCb);
    expect(contentWidth).toEqual('10rem');
    expect(contentHeight).toEqual('10rem');
  });
});

describe('ToggleGroup', () => {
  it('renders with minimal props', () => {
    const { container } = render(
      <ToggleGroup
        id="toggle-group"
        name="test-toggle-group"
        handleToggle={() => {}}
      />,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('calls handleToggle on toggle selection', () => {
    const onToggleCb = jest.fn();
    const { container } = render(
      <ToggleGroup name="test-toggle-group-b" handleToggle={onToggleCb}>
        <TextToggle title="test toggle a" value="a" id="a" />
        <TextToggle title="test toggle b" value="b" id="b" />
      </ToggleGroup>,
    );
    const toggleA = container.querySelector('#a');
    fireEvent.click(toggleA);
    expect(onToggleCb).toHaveBeenCalled();
    expect(onToggleCb.mock.calls[0][0]).toEqual('a');
  });

  it('calls handleClick on toggle click', () => {
    const onToggleCb = jest.fn();
    const onClickCb = jest.fn();
    const { container } = render(
      <ToggleGroup
        name="test-toggle-group-b"
        handleToggle={onToggleCb}
        handleClick={onClickCb}
      >
        <TextToggle label="test toggle a" value="a" id="a" />
        <TextToggle label="test toggle b" value="b" id="b" />
      </ToggleGroup>,
    );
    const toggleA = container.querySelector('#a');
    fireEvent.click(toggleA);
    fireEvent.click(toggleA);
    expect(onToggleCb).toHaveBeenCalledTimes(1);
    expect(onClickCb).toHaveBeenCalledTimes(2);
  });

  it('renders with tooltip', () => {
    const handleChangeCb = jest.fn();
    const tooltip = { title: 'test' };
    const { container } = render(
      <ToggleGroup
        name="test-toggle-group-b"
        handleToggle={handleChangeCb}
        tooltip={tooltip}
      >
        <TextToggle title="test toggle a" id="a" value="val-1" />
        <TextToggle title="test toggle b" id="b" value="val-2" />
      </ToggleGroup>,
    );
    const tooltipWrapper = container.querySelector('[role="tooltip"]');
    expect(tooltipWrapper).toBeInTheDocument();
  });
});
