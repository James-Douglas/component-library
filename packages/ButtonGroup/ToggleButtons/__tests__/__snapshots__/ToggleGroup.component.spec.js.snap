// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleGroup renders with minimal props 1`] = `"<div class=\\"sc-AxheI bXnOQb\\"><label id=\\"\\" for=\\"toggle-group\\" class=\\"sc-Axmtr zreGG label\\"></label></div><div id=\\"toggle-group\\" class=\\"sc-fzpans jdLqgK\\"></div><div class=\\"sc-fzplWN cKZgFO\\"></div>"`;
